_posts/2022-08-05-hacking-for-cruise-control-ui.md

```diff
@@ -0,0 +1,300 @@
+        - name: cluster-ca-certs
+          mountPath: /etc/ssl/certs/
+      volumes:
+        - name: cruise-control-ui
+          configMap:
+            name: cruise-control-ui
+        - name: api-auth-config
+          secret:
+            defaultMode: 292
+            secretName: my-cluster-cruise-control-api
+        - name: cluster-ca-certs
+          secret:
+            defaultMode: 292
+            secretName: my-cluster-cruise-control-certs
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cruise-control-ui
+  labels:
+    app: cruise-control-ui
+data:
+  default.conf: |2+
+        server {
+        listen 9090 ssl;
+        server_name localhost; 
+        ssl_certificate     /etc/ssl/certs/cruise-control.crt;
+        ssl_certificate_key /etc/ssl/certs/cruise-control.key;
+       
+        location / {
+            root   /usr/share/nginx/html;
+            index  index.html index.htm;
+        }
+
+        location /kafkacruisecontrol/ {
+            proxy_set_header Host $host;
+            proxy_set_header X-Real-IP $remote_addr;
+            proxy_buffering off;
+            proxy_pass https://my-cluster-cruise-control:9090/kafkacruisecontrol/;
+            proxy_ssl_verify              off;
+        }
+        
+        error_page   500 502 503 504  /50x.html;
+        location = /50x.html {
+            root   /usr/share/nginx/html;
+        }
+        }
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: cruise-control-ui
+  labels:
+    app: cruise-control-ui
+spec:
+  selector:
+    app: cruise-control-ui
+  ports:
+    - protocol: TCP
+      port: 9090
+      targetPort: 9090
+```
+
+Then run it:
+```
+kubectl apply -f cruise-control-ui.yaml -n <namespace>
+```
+
+## Using Cruise Control Front End
+
+After deployment, the [Cruise Control UI](https://github.com/linkedin/cruise-control-ui) application will be accessible via the cruise-control-ui `service`.
+
+If running the cluster locally with [minikube](https://github.com/kubernetes/minikube), you can port-forward the Cruise Control UI `service` to access it locally:
+
+```
+kubectl port-forward svc/cruise-control-ui 9090:9090
+```
+
+Then navigate to [‘https://127.0.0.1:9090’](https://127.0.0.1:9090) in your browser
+
+![Cruise Control Frontend](/assets/images/posts/2019-08-08-cruise-control-frontend.png)
+
+## Conclusion
+
+Now we have a graphical user interface for Cruise Control running alongside and connected to a Strimzi-managed Kafka cluster. 
+This UI is great for monitoring your Kafka cluster and getting more insight on the data and operations Cruise Control has to offer!
+In one of the future versions, the Strimzi community hopes to add the option for creating custom Cruise Control API users.
+This would remove the requirement for the Cruise Control API security settings to be disabled for applications like the [Cruise Control UI](https://github.com/linkedin/cruise-control-ui) to work with Strimzi but until then, hack away!
```

Comment:
```suggestion
This would remove the requirement for the Cruise Control API security settings to be disabled for applications like the [Cruise Control UI](https://github.com/linkedin/cruise-control-ui) to work with Strimzi. Until then, hack away!
```